### PR TITLE
chore: Fix flags and add tests

### PR DESCRIPTION
**Pull request by jsumners-nr**

### CODE REVIEW

File: main_int_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"os"
+	"path"
+	"strings"
+	"testing"
+)
+import "nrversions"
+
+func Test_RenderCompatDoc(t *testing.T) {
```
Comment (bizob2828) [inline main_int_test.go:13]: this test is fine but i think we need a destructive test. I've found if a target lacks a dep in the tests stanza or a minSupported it fails to parse versions for a package.json

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: main_int_test.go (hunk 1 above)
Comment (bizob2828) [reply main_int_test.go:13]: This is what I did to verify the destructive path

```
diff --git a/testdata/versioned/aws-sdk-v3/package.json b/testdata/versioned/aws-sdk-v3/package.json
index 18468eb..9cf10ca 100644
--- a/testdata/versioned/aws-sdk-v3/package.json
+++ b/testdata/versioned/aws-sdk-v3/package.json
@@ -9,7 +9,7 @@
     {"name": "@aws-sdk/client-dynamodb", "minAgentVersion": "8.7.1"},
     {"name": "@aws-sdk/lib-dynamodb", "minAgentVersion": "8.7.1"},
     {"name": "@aws-sdk/smithy-client", "minSupported": "3.47.0", "minAgentVersion": "8.7.1"},
-    {"name": "@smithy/smithy-client", "minSupported": "2.0.0", "minAgentVersion": "11.0.0"},
+    {"name": "@smithy/smithy-client", "minAgentVersion": "11.0.0"},
     {"name": "@aws-sdk/client-bedrock-runtime", "minAgentVersion": "11.13.0"}
   ],
   "version": "0.0.0",
```

Comment (jsumners-nr) [reply main_int_test.go:13]: Yes, this is an original design decision. When we can't find a version, we assume the whole manifest is broken and bail on it:

https://github.com/newrelic/newrelic-node-versions/blob/023ae857a5a4aab890a0aed87af52d4ae4066374/parse-package.go#L36-L52

Comment (jsumners-nr) [reply main_int_test.go:13]: I'd say that changing that behavior is outside of the scope of this PR. It is a definite change in the way the tool works.

Comment (bizob2828) [reply main_int_test.go:13]: that's fair

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**